src/opera/parser/tosca/csar.py

```diff
@@ -1,19 +1,27 @@
+import shutil
+import yaml
+
 from pathlib import Path
 from tempfile import TemporaryDirectory
 from zipfile import ZipFile
 
-import yaml
 from opera.error import ParseError
 
 
 class CloudServiceArchive:
-    def __init__(self, csar_name, csar_folder_path):
+    def __init__(self, csar_name, csar_folder_path=None):
         self._csar_name = csar_name
         self._csar_folder_path = csar_folder_path
         self._tosca_meta = None
         self._root_yaml_template = None
         self._metadata = None
 
+    def unpackage_csar(self, output_dir, csar_format="zip"):
+        # validate CSAR before unpacking it
+        self.validate_csar()
```

Comment:
Hmm okay, I agree that validation and unpackaging can be seen as separated operations. But can they really be separated? I see validation as a necessary thing here. Because without it the purpose of `opera unpackage` can become too general since it would extract every zip or tar compressed archive to specified folder without even knowing if this is a valid CSAR or not. This is the only second thought that I have right know.

Reply:
This can be resolved - the validation was moved out from the `csar.py` to the `unpackage.py` which is better.